cmd/oras/internal/option/pusher.go

```diff
@@ -60,10 +67,21 @@ func (opts *Pusher) LoadManifestAnnotations() (map[string]map[string]string, err
 		if err := decodeJSON(opts.ManifestAnnotations, &annotations); err != nil {
 			return nil, err
 		}
+		if len(annotations[annotationManifest]) == 0 && len(opts.FileRefs) == 0 {
+			return nil, fmt.Errorf("no blob and manifest annotation file %s is empty", opts.ManifestAnnotations)
+		}
 	}
 	return annotations, nil
 }
 
+// ValidateEmpty checks whether blobs or manifest annotation are empty.
+func (opts *Pusher) ValidateEmpty(args []string) error {
+	if len(args) == 1 && opts.ManifestAnnotations == "" {
+		return errors.New("no blob and manifest annotation are provided")
+	}
+	return nil
+}
```

Comment:
This logic is `attach` specific. You might want to take it out of the `Pusher` option.